src/main/kotlin/org/rust/ide/completion/AttributeCompletionProvider.kt

```diff
@@ -0,0 +1,164 @@
+import com.intellij.codeInsight.completion.CompletionResultSet
+import com.intellij.codeInsight.lookup.LookupElementBuilder
+import com.intellij.patterns.ElementPattern
+import com.intellij.patterns.PlatformPatterns.psiElement
+import com.intellij.patterns.PlatformPatterns.virtualFile
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiFile
+import com.intellij.util.ProcessingContext
+import org.rust.lang.RustLanguage
+import org.rust.lang.core.psi.*
+
+object AttributeCompletionProvider : CompletionProvider<CompletionParameters>() {
+
+    private val attributes = listOf(
+        RustAttribute("crate_name", onCrate),
+        RustAttribute("crate_type", onCrate),
+        RustAttribute("feature", onCrate),
+        RustAttribute("no_builtins", onCrate),
+        RustAttribute("no_main", onCrate),
+        RustAttribute("no_start", onCrate),
+        RustAttribute("no_std", onCrate),
+        RustAttribute("plugin", onCrate), // Feature gated as of 1.9
+        RustAttribute("recursion_limit", onCrate),
+        RustAttribute("no_implicit_prelude", onMod),
+        RustAttribute("path", onMod),
+        RustAttribute("main", onFn),
+        RustAttribute("plugin_registrar", onFn),
+        RustAttribute("start", onFn),
+        RustAttribute("test", onFn),
+        RustAttribute("should_panic", onTestFn),
+        RustAttribute("cold", onFn),
+        RustAttribute("naked", onFn),
+        RustAttribute("thread_local", onStaticMut),
+        RustAttribute("link_args", onExternBlock), // Feature gated as of 1.9
+        RustAttribute("link", onExternBlock),
+        RustAttribute("linked_from", onExternBlock), // Feature gated as of 1.9
+        RustAttribute("link_name", onExternBlockDecl),
+        RustAttribute("linkage", onExternBlockDecl),
+        RustAttribute("repr", onStruct or onEnum),
+        RustAttribute("macro_use", onMod or onExternCrate),
+        RustAttribute("macro_reexport", onExternCrate),
+        RustAttribute("macro_export", onMacro),
+        RustAttribute("no_link", onExternCrate),
+        RustAttribute("export_name", onStatic or onFn),
+        RustAttribute("link_section", onStatic or onFn),
+        RustAttribute("no_mangle", onAnyItem),
+        RustAttribute("simd", onTupleStruct), // Feature gated as of 1.9
+        RustAttribute("unsafe_destructor_blind_to_params", onDropFn), // Feature gated as of 1.9
+        RustAttribute("unsafe_no_drop_flag", onStruct), // Feature gated as of 1.9
+        RustAttribute("doc", onAnyItem),
+        RustAttribute("rustc_on_unimplemented", onTrait), // Feature gated as of 1.9
+        RustAttribute("cfg", onFn),
+        RustAttribute("cfg_attr", onAnyItem),
+        RustAttribute("allow", onAnyItem),
+        RustAttribute("warn", onAnyItem),
+        RustAttribute("forbid", onAnyItem),
+        RustAttribute("deny", onAnyItem),
+        RustAttribute("lang", onFn),
+        RustAttribute("inline", onFn),
+        RustAttribute("derive", onStruct or onEnum))
```

Comment:
Thanks :) Being able to define arbitrary infix functions is pretty neat.  I'm really enjoying Kotlin.